### ISSUE TITLE
Stroke color not set for Python3 version of "_scene2.line" 

### ISSUE THREAD

**Issue by RichardPotthoff** [omz/Pythonista-Issues#149]
When I tried to convert an old Python 2 game to Python 3 (The game was Guy Carver's ripoff.py, https://gist.github.com/GuyCarver/4115763), I had problems getting the line colors to work. All lines had the color of a rectangle drawn earlier. I was able to get everything to work by drawing a dummy rectangle before drawing each line:
<pre><code>def drawline(p1, p2): 
	rect(0,0,0,0) # <-dummy call to get the stoke color working for line()
	line(p1.x, p1.y, p2.x, p2.y)
</code></pre>

I also noticed that the default argument values for <code>_screne_types.Color</code> are not working. This means I get error messages if not all 4 arguments are supplied:
<pre><code>    c=_scene_types.Color()
TypeError: __init__() missing 4 required positional arguments: 'r', 'g', 'b', and 'a'
</code></pre>

**Comment by Dutcho** [comment]
The root cause of above issue is the `recordtype` function in `site-packages/_scene_types.py`, which wasn't fully updated from Python 2 to 3. This causes **default handling** not to work for _any_ `recordtype`, so the same issue applies to `Vector3`; others (i.e. `Point`, `Size`, `Rect`) aren't imported (anymore) from `_scene_types` by `scene_drawing`; the latter is fully imported (`import * from scene_drawing`) by `scene.py`. This issue is also demonstrated by the `doctest` on line 23 `>>> Point()`, which fails on `missing 2 required positional arguments`.
To correct this, I believe lines 137-138 should change to the (unconditional) line:
```
    cls.__init__.__defaults__ = init_defaults
```

**Comment by RichardPotthoff** [comment]
Fixing the <code>_scene_types.recordtype</code> bug did not fix the <code>\_scene2.line</code> problem I have.

Here is a link to the comparison between the original Python2 version of the game, and the modifications I had do make to get it running under Python3 in Pythonista3v3.0(300018):

https://github.com/RichardPotthoff/myrepo/compare/050744...7271d1

I also had a patched version of <code>_scene_types.py</code> (see issue #150) in <code>site-packages3</code>, but I suspect that <code>_scene2</code> does not load modules from there.

Here is a screenshot of the game without the line <code>120 +  rect(0,0,0,0)</code>:

![image](https://cloud.githubusercontent.com/assets/1579528/17675901/d80fe9e4-62f9-11e6-82c0-7b4f715b3e40.png)

And here the same with the dummy rectangle drawn before each <code>_scene2.line</code>call:

![image](https://cloud.githubusercontent.com/assets/1579528/17675902/d8269446-62f9-11e6-941b-33c67efd85ca.png)

**Comment by Dutcho** [comment]
Ah, so the broken default handling wasn't the issue, but the need for the need for the `rect(0,0,0,0)` workaround was?

**Comment by RichardPotthoff** [comment]
Exactly! Well, allmost: My initial workaround for the <code>Color</code> problem was to simply add the missing 4th argument to get rid of the errors, but then I found the bug in the <code>_scene_type.recordtype</code> class factory function, and posted a separate bug report (issue #150) with my patch. 
I was not able to fix the <code>_scene2.line</code> bug, because I do not have the code for the <code>_scene2</code> module, so I had to keep the <code>rect(0,0,0,0)</code> workaround for that problem.

**Comment by Dutcho** [comment]
The issue doesn't occur in a minimal classic render loop scene script, so _somewhere_ in `ripoff.py` _something_ breaks `line`, which `rect` fixes then.
I don't know `ripoff.py`, and it's quite big, but it seems out-of-lining the `rect` workaround from the `drawline` function to the main `MyScene.draw` function has the same beneficial effect, but only if inserted after drawing the crates. Don't know, but perhaps that helps finding the root cause

**Comment by Dutcho** [comment]
And indeed `line` and `rect` are defined in `_scene2`, which is built-in and hence inaccessible
